File: TP1/Exo 1.6/python/triangle.py
def triangle(taille):
    # Validation pour vérifier que la taille est strictement inférieure à 10
    if taille >= 10:
        print("La taille doit être strictement inférieure à 10.")
        return  # Arrête la fonction si la condition n'est pas remplie
    
    # Boucle pour dessiner le triangle
    for i in range(1, taille + 1):  # Parcourt chaque ligne du triangle
        line = ""  # Initialisation de la ligne à afficher
        for j in range(i):  # Parcourt chaque position dans la ligne courante
            # Affichage des contours avec '*'
            if j == 0 or j == i - 1 or i == taille:  # Premier ou dernier caractère de la ligne, ou si c'est la dernière ligne
                line += "* "  # Ajoute '*' à la ligne
            else:
                # Affichage de l'intérieur avec '#'
                line += "# "  # Ajoute '#' à la ligne pour les positions intérieures
        print(line.strip())  # Affiche la ligne en supprimant les espaces supplémentaires à la fin

def main():
    # Demande à l'utilisateur d'entrer une taille pour le triangle
    taille = int(input("Entrez une valeur de taille de triangle inférieure à 10 : "))
    # Appelle la fonction triangle avec la taille donnée
    triangle(taille)

# Point d'entrée du programme
if __name__ == "__main__":
    main()

File: TP1/Exo 1.6/C/triangle.c
#include <stdio.h> // Inclusion de la bibliothèque standard pour les entrées/sorties

// Fonction pour dessiner un triangle
void triangle(int taille)
{
    // Boucle pour chaque ligne du triangle, de 1 à la taille donnée
    for (int i = 1; i <= taille; i++)
    {
        // Boucle pour chaque caractère à afficher sur la ligne i
        for(int j = 0; j < i; j++)
        {
            // Conditions pour savoir quand afficher un "*" ou un "#"
            if (j == 0 || j == i - 1 || i == taille)
            {
                printf("* ");  // Affiche "*" si on est sur la bordure du triangle ou sur la dernière ligne
            }
            else
            {
                printf("# ");  // Affiche "#" si on est à l'intérieur du triangle, sans bordure
            }
        }
        printf("\n"); // Nouvelle ligne pour passer à la ligne suivante du triangle
    }
}

int main()
{
    int taille;

    // Demande à l'utilisateur de saisir une taille pour le triangle
    printf("Entrez une valeur pour la taille (strictement inférieure à 10) : ");
    // Vérifie que l'utilisateur entre un entier valide entre 1 et 9
    while (scanf("%d", &taille) != 1 || taille >= 10 || taille < 1) {
        printf("Veuillez entrer un entier valide entre 1 et 9 : ");
        while (getchar() != '\n'); // Vide le buffer d'entrée en cas d'entrée non valide
    }

    // Appelle la fonction triangle pour dessiner le triangle de la taille donnée
    triangle(taille);

    return 0; // Fin du programme
}
